src/compiler/diagnosticMessages.json

```diff
@@ -1652,6 +1652,10 @@
         "category": "Error",
         "code": 2525
     },
+    "'this' type is available only in a non-static member of a class or interface.": {

```

Comment:
No, this isn't about type annotations, this is about references to the `this` type (which might or might not occur in type annotations, for example the reference could be in a type assertion).